src/ripple/app/tx/impl/applySteps.cpp

```diff
@@ -50,371 +50,226 @@
         case ttTICKET_CREATE:
-            return invoke_preflight_helper<CreateTicket>(ctx);
+            return f.template operator()<CreateTicket>();
         case ttTRUST_SET:
-            return invoke_preflight_helper<SetTrust>(ctx);
+            return f.template operator()<SetTrust>();
         case ttAMENDMENT:
         case ttFEE:
         case ttUNL_MODIFY:
-            return invoke_preflight_helper<Change>(ctx);
+            return f.template operator()<Change>();
         case ttNFTOKEN_MINT:
-            return invoke_preflight_helper<NFTokenMint>(ctx);
+            return f.template operator()<NFTokenMint>();
         case ttNFTOKEN_BURN:
-            return invoke_preflight_helper<NFTokenBurn>(ctx);
+            return f.template operator()<NFTokenBurn>();
         case ttNFTOKEN_CREATE_OFFER:
-            return invoke_preflight_helper<NFTokenCreateOffer>(ctx);
+            return f.template operator()<NFTokenCreateOffer>();
         case ttNFTOKEN_CANCEL_OFFER:
-            return invoke_preflight_helper<NFTokenCancelOffer>(ctx);
+            return f.template operator()<NFTokenCancelOffer>();
         case ttNFTOKEN_ACCEPT_OFFER:
-            return invoke_preflight_helper<NFTokenAcceptOffer>(ctx);
+            return f.template operator()<NFTokenAcceptOffer>();
         case ttCLAWBACK:
-            return invoke_preflight_helper<Clawback>(ctx);
+            return f.template operator()<Clawback>();
         case ttAMM_CREATE:
-            return invoke_preflight_helper<AMMCreate>(ctx);
+            return f.template operator()<AMMCreate>();
         case ttAMM_DEPOSIT:
-            return invoke_preflight_helper<AMMDeposit>(ctx);
+            return f.template operator()<AMMDeposit>();
         case ttAMM_WITHDRAW:
-            return invoke_preflight_helper<AMMWithdraw>(ctx);
+            return f.template operator()<AMMWithdraw>();
         case ttAMM_VOTE:
-            return invoke_preflight_helper<AMMVote>(ctx);
+            return f.template operator()<AMMVote>();
         case ttAMM_BID:
-            return invoke_preflight_helper<AMMBid>(ctx);
+            return f.template operator()<AMMBid>();
         case ttAMM_DELETE:
-            return invoke_preflight_helper<AMMDelete>(ctx);
+            return f.template operator()<AMMDelete>();
         case ttXCHAIN_CREATE_BRIDGE:
-            return invoke_preflight_helper<XChainCreateBridge>(ctx);
+            return f.template operator()<XChainCreateBridge>();
         case ttXCHAIN_MODIFY_BRIDGE:
-            return invoke_preflight_helper<BridgeModify>(ctx);
+            return f.template operator()<BridgeModify>();
         case ttXCHAIN_CREATE_CLAIM_ID:
-            return invoke_preflight_helper<XChainCreateClaimID>(ctx);
+            return f.template operator()<XChainCreateClaimID>();
         case ttXCHAIN_COMMIT:
-            return invoke_preflight_helper<XChainCommit>(ctx);
+            return f.template operator()<XChainCommit>();
```

Comment:
I had not come across the usage of the `template` keyword, I learnt a new construct :) thanks.